packages/react/src/hooks/useIsChildFocused.ts

```diff
@@ -0,0 +1,49 @@
+import {useCallback, useEffect, useRef, useState, type RefObject} from 'react'
+
+/**
+ * Determine if a child element of the provided ref is currently focussed.
+ * @param onFocusChange The callback to be called when the focus state changes.
+ * @type T The type of the container element.
+ * @returns The ref to be applied to the container element.
+ */
+export const useIsChildFocused = <T extends HTMLElement>(
```

Comment:
Are you seeing a lot of reuse of this hook @joshfarrant, to warrant it being being hoisted from the source folder of sub nav? 

I think its use-cases are a little limited, so if we're keeping this I'd be in favor of co-locating it to the sub nav folder and keeping it private. We do this for some other components like anchor nav. 

Arguably, you could also eliminate most of the lines here and put the useEffect back into SubNav? I think that's the best option, given there's only one effect and the state management is incidental.

Reply:
Happy to co-locate. I think I had it alongside SubNav before but thought I'd pull it up as it doesn't do anything specific to SubNav. I'll move it back into the directory 👍

I'd advocate for keeping it as its own hook, purely because it groups all of the logic related to this functionality together into a single place, along with a descriptive name. What lines are you thinking could be removed?